src/main/java/frc/robot/poseestimator/WPILibPoseEstimator/WPILibPoseEstimator.java

```diff
@@ -0,0 +1,115 @@
+import java.util.Optional;
+
+public class WPILibPoseEstimator extends GBSubsystem implements IPoseEstimator {
+
+	private final PoseEstimator<SwerveModulePosition[]> poseEstimator;
+	private final Odometry<SwerveModulePosition[]> odometryEstimator;
+
+	private final TimeInterpolatableBuffer<Pose2d> timeInterpolatableBuffer;
+
+	public WPILibPoseEstimator(
+		String logPath,
+		SwerveDriveKinematics kinematics,
+		SwerveDriveOdometry odometry,
+		SwerveModulePosition[] modulePositions
+	) {
+		super(logPath);
+
+
+		this.poseEstimator = new PoseEstimator<>(
+			kinematics,
+			odometry,
+			WPILibPoseEstimatorConstants.DEFAULT_ODOMETRY_STANDARD_DEVIATIONS.getWPILibStandardDeviations(),
+			WPILibPoseEstimatorConstants.DEFAULT_VISION_STANDARD_DEVIATIONS.getWPILibStandardDeviations()
+		);
+		this.odometryEstimator = new Odometry<>(
+			kinematics,
+			WPILibPoseEstimatorConstants.STARTING_ODOMETRY_ANGLE,
+			modulePositions,
+			WPILibPoseEstimatorConstants.STARTING_ODOMETRY_POSE
+		);
+
+		this.timeInterpolatableBuffer = TimeInterpolatableBuffer.createBuffer(WPILibPoseEstimatorConstants.POSE_BUFFER_SIZE_SECONDS);
+		;
+	}
+
+
+	@Override
+	public void resetPose(Pose2d newPose) {
+		poseEstimator.resetPose(newPose);
+	}
+
+	@Override
+	public Pose2d getEstimatedPose() {
+		return poseEstimator.getEstimatedPosition();
+	}
+
+	@Override
+	public Pose2d getEstimatedPoseAtTimestamp(double timestamp) {
+		return timeInterpolatableBuffer.getSample(timestamp).orElseGet(this::getEstimatedPose);
+	}
+
+	@Override
+	public void updateOdometry(OdometryObservation[] odometryObservations) {
+		for (OdometryObservation odometryObservation : odometryObservations) {
+			poseEstimator.update(odometryObservation.gyroAngle(), odometryObservation.wheelPositions());
+			updateOdometryPose(odometryObservation);
+		}
+	}
+
+	@Override
+	public void resetOdometry(SwerveModulePosition[] wheelPositions, Rotation2d gyroAngle, Pose2d robotPose) {
+		poseEstimator.resetPosition(gyroAngle, wheelPositions, robotPose);
+		odometryEstimator.resetPosition(gyroAngle, wheelPositions, robotPose);
+	}
+
+	@Override
+	public Pose2d getOdometryPose() {
+		return odometryEstimator.getPoseMeters();
+	}
+
+	@Override
+	public void setHeading(Rotation2d newHeading) {
+		poseEstimator.resetRotation(newHeading);
+		odometryEstimator.resetRotation(newHeading);
+	}
+
+	@Override
+	public void updateVision(List<IRobotPoseVisionObservation> robotPoseVisionData) {}
+
+	@Override
+	public Optional<Pose2d> getVisionPose() {
+		return Optional.empty();
+	}
+
+	private void updateOdometryPose(OdometryObservation observation) {
+		odometryEstimator.update(observation.gyroAngle(), observation.wheelPositions());
+	}
+
+	private void addVisionMeasurement(RawAprilTagVisionData visionObservation) {
+		poseEstimator.addVisionMeasurement(
+			visionObservation.getEstimatedPose().toPose2d(),
+			visionObservation.getTimestamp(),
+			VecBuilder.fill(0, 0, 0) // todo change to funciton that calculats
```

Comment:
No todo, the whole point of a simple WPIlib PoseEstimator is that it is simple, with constant stdDevs that do not depend on a function that could break at competition.